drivers/usb/device/usb_dc_stm32.c

```diff
@@ -125,29 +184,90 @@ static int usb_dc_stm32_clock_enable(void)
 {
 	struct device *clk = device_get_binding(STM32_CLOCK_CONTROL_NAME);
 	struct stm32_pclken pclken = {
+
+#ifdef USB
+		.bus = STM32_CLOCK_BUS_APB1,
+		.enr = LL_APB1_GRP1_PERIPH_USB,
+#else /* USB_OTG_FS */
+
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+		.bus = STM32_CLOCK_BUS_AHB1,
+		.enr = LL_AHB1_GRP1_PERIPH_OTGFS,
+#else
 		.bus = STM32_CLOCK_BUS_AHB2,
 		.enr = LL_AHB2_GRP1_PERIPH_OTGFS,
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+
+#endif /* USB */
 	};
 
 	clock_control_on(clk, (clock_control_subsys_t *)&pclken);
 
 	return 0;
 }
 
+#ifdef USB
```

Comment:
Seems this function could be refactored , no ?

Reply:
Which function? usb_dc_stm32_clock_enable or usb_dc_stm32_init?

Reply:
usb_dc_stm32_init()

Reply:
Are you talking about merging the two init functions of the usb controllers in one and use ifdefs for the differences inside the init?

Reply:
Yes

Reply:
Done